### DISCUSSION TITLE
.net faster execution and modern challenges

### DISCUSSION THREAD

**Discussion by IrakliLomidze** [dotnet/csharplang#6246] [Language Ideas]
**Background and motivation**
The silicon-based CPUs are nearly approaching their physical limits, Abd the Quantum computers have not arrived yet. It means that code execution performance becomes Business-critical again. Large portions of code in the world are written in managed languages (such as java and .net). In comparison with a native code, they are slower.

Please add more native execution capability in .net (such as reference counting, non-reflective classes, non-exception blocks, non-garbage collatable classes, and more).

**Idea 1:** The ability to manually allocate a class and keep it until we need it and then manually destroy it. C# has similar functionality since the early version fixed keyword, but it works for byte array only.
**Scenario 1.** We read fixed-length records from the database one by one. We manipulate data and save it back to the database. Expensive operations: Manipulation means Recreate entity each time when we change data (e.g. string). Each time, we are allocating new data into the heap. We might want to reuse existing allocating memory for those records. For that, we need to have:
1) **fixed** keyword for class and struct.
2) **stack** keyword to allocate some data into the stack
3) **Metaless** keyword when we sometimes absolutely don’t need metadata with the class

**Idea 2**: Sometimes we have code that requires high performance, but we have to deal with super expensive exceptions. Scenario 2. We have a Web API with multiple fallback endpoints to call some web API. We call them one by one, and until the call is successful. Each time we get expensive exceptions with stack info. However, we only need status and error codes. I understand that C# is designed to raise exceptions. In C++ we have **noexceptxecp** keyword. But it will be nice that C# has specified what they should exception include. **minexcept** keyword

**Comment by CyrusNajmabadi** [comment]
@IrakliLomidze can you please open individual discussions for the particular language changes you want to see.  One large discussion is not going to be conducive as it will be covering too many topics.

**Comment by HaloFour** [comment]
Most of this seems like [runtime](https://github.com/dotnet/runtime) asks, as the runtime does not offer the functionality that you're describing.

**Reply by svick** [reply]
This request was originally posted as https://github.com/dotnet/runtime/issues/71299, where it was closed in part as being more appropriate here.

**Reply by HaloFour** [reply]
I guess that depends on whether or not language keywords that depend on runtime functionality that doesn't exist is a language issue or a runtime issue.  😄
